Самойлов Владимир

patrn1 opened the pull request.

Commit:
Update phone-book.js

Commit:
Update phone-book.js

Commit:
Update phone-book.js

Commit:
Update phone-book.js

Commit:
Update phone-book.js

i4got10 commented on phone-book.js:
```diff
@@ -1,52 +1,187 @@
+        return;
+    }
+    
+    var invalidPhones = [
```
Лучше не перебирать все неправильные варианты, а перебирать правильные

Commit:
Update phone-book.js

Commit:
Edit reg exp

Commit:
Fix remove & find functions

i4got10 commented:
Если выполнить .find() без параметров, то функция ничего не возвращает

Commit:
Fix find function

i4got10 commented:
Телефон должен выводится в отформатированном виде. Например, 
`Dane Hinrichs, +7 (111) 777 2 222, info@example.com` => `Dane Hinrichs, +7 (111) 777-2-222, info@example.com`

Commit:
Add phone number formatting

i4got10 commented:
1) Не прошли такие телефоны(это из задания)

```
111 777-2-222
71117772222
```

2) Поиск по части номера не работает `phoneBook.find('+7 (999)');`
3) Удаление по номеру не работает `phoneBook.remove('+7 (999) 666-7-778');`

Commit:
Change find & remove functions

Usage:
phoneBook.find('+7 (999) 111-1-111');
phoneBook.remove('+7 (999) 111-1-111');
phoneBook.find('79991111111');
phoneBook.remove('79991111111');

i4got10 commented:
Вот эти тесты не прошли
![2015-10-13_1347](https://cloud.githubusercontent.com/assets/1151154/10450029/ee4bf12a-71b0-11e5-9915-32e1aa1260dd.png)

i4got10 commented on phone-book.js:
```diff
@@ -1,52 +1,229 @@
+        return;
+    }
+    
+    var validPhone = /^(\+?\s*\d+\s*)?(\(\d+\)|\d+)(\s*\-?\s*\d+){1,}$/;;
```
;;

i4got10 commented on phone-book.js:
```diff
@@ -1,52 +1,229 @@
+    var invalidEmails = [
+        "^\\.|\\.@|\\.\\.",
+        "@.+-\\.|@-"
+        ];
```
отступ лишний

i4got10 commented on phone-book.js:
```diff
@@ -1,52 +1,229 @@
+    var emptyTableRow = '▓' + 
+        Array(nameColumnWidth  + 2).join(' ') + ' │' +
+        Array(phoneColumnWidth + 2).join(' ') + ' │' +
+        Array(emailColumnWidth + 2).join(' ') + ' ▓';
```
`Array(emailColumnWidth + 2).join(' ')` можно сделать отдельной функцией более красиво

v-samoylov replied:
в задании же сказано:
>add(name, phone, email) – добавляет запись в телефонную книгу, если какое-либо из полей не валидно, то запись не добавляется;

Commit:
Update phone-book.js

i4got10 commented:
:star: Отлично!
